cms/bundles/forms.py

```diff
@@ -82,9 +82,10 @@ def clean(self) -> dict[str, Any] | None:
                 if self.instance.created_by_id == self.for_user.pk:
                     cleaned_data["status"] = self.instance.status
                     self.add_error("status", ValidationError("You cannot self-approve your own bundle!"))
-
-                cleaned_data["approved_at"] = timezone.now()
-                cleaned_data["approved_by"] = self.for_user
+                else:
+                    # the approver is different from the creator, so let's populate the relevant fields.
+                    cleaned_data["approved_at"] = timezone.now()
+                    cleaned_data["approved_by"] = self.for_user
```

Comment:
🙈 sorry to be a pain: This implies we don't have a unit test to ensure that these values are only updated if it is not self-approved.

Reply:
👍🏼  6c3afbeec